osi_sensordata.proto

```diff
@@ -60,143 +68,137 @@ message SensorData
     //
-    optional MountingPosition mounting_position_rmse = 7;
+    optional MountingPosition mounting_position_rmse = 5;
+
+    // Sensor view w.r.t. the sensor coordinate system
+    //
+    // This provides a copy of the \c SensorView data received by the sensor
+    // for reference purposes.
+    //
+    optional SensorView sensor_view = 6;
 
     // The timestamp of the last real-world measurement (e.g. GT input) that
     // this set of sensor data takes into account.  This in effect is the last
     // time instance of reality the measurements correspond to.  See field
     // \c SensorData::timestamp for a detailed discussion.  This value is also
     // the upper bound to the \c DetectedObjectHeader::measurement_time and the
     // feature data \c SensorDetectionHeader::measurement_time fields.
-    // 
-    optional Timestamp last_measurement_time = 8;
+    //
+    optional Timestamp last_measurement_time = 7;
 
     // General information about the \c DetectedObject .
-    // 
-    optional DetectedEntityHeader object_header = 100;
-    
+    //
+    optional DetectedEntityHeader object_header = 8;
+
     // The list of objects detected by the sensor as perceived by the sensor.
     //
     repeated DetectedObject object = 9;
-    
+
     // General information about the \c DetectedTrafficSign .
-    // 
-    optional DetectedEntityHeader traffic_sign_header = 101;
+    //
+    optional DetectedEntityHeader traffic_sign_header = 10;
 
     // The list of traffic signs detected by the sensor
     //
-    repeated DetectedTrafficSign traffic_sign = 10;
-        
+    repeated DetectedTrafficSign traffic_sign = 11;
+
     // General information about the \c DetectedTrafficLight .
-    // 
-    optional DetectedEntityHeader traffic_light_header = 102;
+    //
+    optional DetectedEntityHeader traffic_light_header = 12;
 
     // The list of traffic lights detected by the sensor
-    // 
-    repeated DetectedTrafficLight traffic_light = 11;
+    //
+    repeated DetectedTrafficLight traffic_light = 13;
 
     // General information about the \c DetectedRoadMarking .
-    // 
-    optional DetectedEntityHeader road_marking_header = 103;
+    //
+    optional DetectedEntityHeader road_marking_header = 14;
 
     // The list of road marking detected by the sensor.
     // This excludes lane boundary markings.
     //
-    repeated DetectedRoadMarking road_marking = 12;
-        
+    repeated DetectedRoadMarking road_marking = 15;
+
     // General information about the \c DetectedLane .
-    // 
-    optional DetectedEntityHeader lane_header = 104;
+    //
+    optional DetectedEntityHeader lane_header = 16;
 
     // The list of lane detected by the sensor
-    // 
-    repeated DetectedLane lane = 13;
-        
+    //
+    repeated DetectedLane lane = 17;
+
+    // The ID of the lane the host vehicle travels on in the DetectedLane data,
+    // relative to the sensor.
+    //
+    optional Identifier host_vehicle_lane_id = 18;
```

Comment:
Where is the host_vehicle_data gone? Shouldn't this be part of it?

Reply:
found it! Isn't this some "self-localization information" being part of the HostVehicleData? What do you think?

Reply:
The Host Vehicle Data is only contained in SensorView, since it is data that the vehicle "knows" about itself, and not something that a sensor would output on its own (we currently have no support for modeling e.g. GPS receivers as sensor models, i.e. producing SensorData). Somewhere along the further processing tool-chain one would imagine other modules taking SensorData and GPS and other data into account to update this internal information.

If one wanted to give sensor models (or fusion models) the ability to express an opinion on the vehicle location, then this should be a new field (estimated host vehicle location, or something of this sort), since host vehicle data could at some point in time contain e.g. brake pedal position, etc., which is not something that ADAS sensors have any opinions about.

Reply:
If you look in the osi_hostvehicledata.proto you find that it is defined to be the place for an "estimated position"!
It is not the Interface for the brake pedal atm but in fact the one for the GPS, assigned lane, etc.... it is opinion only? :dancer: 
I think it must be part of SensorData

Reply:
Given that I wrote that: Sure it is an estimated position, since the vehicle itself has no way of knowing the proper true position. But as stated in that file's header:

    // \brief Interface for host vehicle data that is available to sensors and
    // other functions due to host vehicle's internal communication.

I.e. this is the input side that's available to sensors due to internal communication, it was not meant for the output side of a single sensor. It is the current best knowledge of the vehicle itself, like its knowledge of its break pedal position, RPMs, etc.

I think any estimation by a single sensor in SensorData should follow the pattern for all other detected output we have (i.e. it should be added as single attributes to SensorData). I'll push a commit for that.